src/accessors/IConfigurationExtend.ts

```diff
@@ -0,0 +1,11 @@
+import { ISetting } from '../settings/index';
+
+/**
+ * This accessor provides methods for accessing
+ * settings in a read-only-fashion.
+ */
+export interface ISettingRead {
```

Comment:
@graywolf336 this looks a bit ugly. I thought I branched from master. No clue why this is included in this PR...